### PR TITLE
Add some logging and tests related to sms reminder

### PR DESCRIPTION
**Pull request by nikomakela**
PT-1707 PT-1708 PT-1601

### CODE REVIEW

**Comment by terovirtanen** [comment]
**KULTUS-API branch is deployed to platta: https://kultus-pr332.api.dev.hel.ninja** :rocket::rocket::rocket:

File: occurrences/models.py [hunk 1]
```diff
@@ -813,18 +813,21 @@ def send_enrolment_summary_report_to_providers(cls, days=1):
         # Send enrolment summary
         occurrences_services.send_enrolment_summary_report_to_providers(enrolments)
 
+    def get_contact_people(self) -> list[Person]:
```
Comment (karisal-anders) [inline occurrences/models.py:816]: Although we're using Python 3.9 which supports list[Person] I'm not sure whether some other part in the CI/CD pipeline doesn't as it has [broken](https://dev.azure.com/City-of-Helsinki/kultus/_build/results?buildId=143290&view=logs&j=8267e077-3a31-5bba-93c5-4b3e55896ad0&t=45ce9a3b-1c36-5c65-31cd-c5d7e373a2ce) on this line:
```
 def get_contact_people(self) -> list[Person]:
TypeError: 'type' object is not subscriptable
```

Comment (nikomakela) [reply occurrences/models.py]: yea, dunno 😞 Changed it to `typing.List` now.

**Comment by nikomakela** [comment]
I also thought that I already fixed the issue of notifications being sent in a wrong order, but it seems like I didn't. Maybe, but just maybe, the migration changes in https://github.com/City-of-Helsinki/palvelutarjotin/pull/333 could change something to it.

**Review by karisal-anders — APPROVED** [review]
Test case could perhaps be improved by splitting it into two parts, the one with consent given and the one with no consent given and moving the shared factory using part to own function. As long as you fix the snapshots and the tests pass, ok to merge!

**Comment by terovirtanen** [comment]
**KULTUS-API branch is deployed to platta: https://kultus-pr332.api.dev.hel.ninja** :rocket::rocket::rocket:

**Pull request merged**